Fixed the loading of JS to use the ResourceLoader MediaWiki standard

slim16165 opened the pull request:
After playing for some time with MW, I've understood there was an issue with the direct call to the JS files and with the parsing of " in PHP. Replacing the call with ResourceLoader worked.

https://www.tematichedigenere.com/wiki/Speciale:SemanticMediaWikiGraph?debug=true

gianluigi-salvi-lottomatica committed:
Minimal changes

Added gitignore

Create dependency-review.yml

Create main.yml and dependency-review.yml

gianluigi-salvi-lottomatica committed:
* Added dependencies, extension.json and package.json

gianluigi-salvi-lottomatica committed:
Attempts to load mw object in Javascript

gianluigi-salvi-lottomatica committed:
Attempts to load the custom Javascript using modules and ResourceLoader

gianluigi-salvi-lottomatica committed:
Attempts to load the custom Javascript using modules and ResourceLoader [it was already done in PHP, my extension.json was wrong]

gianluigi-salvi-lottomatica committed:
Work in progress

gianluigi-salvi-lottomatica committed:
Compacted JS Syntax through P42.js (nothing special)

gianluigi-salvi-lottomatica committed:
Conversion to TypeScript using also VisualStudio

gianluigi-salvi-lottomatica committed:
Conversion to TypeScript compiling

gianluigi-salvi-lottomatica committed:
Conversion to TypeScript compiling

krabina commented:
Thank you for this. I did not have any plans on working in this, just wanted to make sure the code stays available. 

But I appreceate your effort in keeping this extension alive! Further commits are most welcome! As soon as we know it is working again, we might as well hand it over to the Semantic MediaWiki repo.

slim16165 commented:
> Thank you for this. I did not have any plans on working in this, just wanted to make sure the code stays available.
> 
> But I appreceate your effort in keeping this extension alive! Further commits are most welcome! As soon as we know it is working again, we might as well hand it over to the Semantic MediaWiki repo.

Thanks a lot! Now I'm working on a major update. I've converted the JS to TypeScript, from procedural to OOP, updated the dependencies to the latest version and tried to comply with the single responsability principle.
